## Atividade 001

Use a planilha de dados compartilhada pelo professor e calcule o IC 90%, 95% e 99%. Compare a largura dos intervalos.

* Suponha que o SLA da companhia seja até 15 minutos de embarque em média.

* O valor 15 está dentro do IC? O que isso sugere?

* Remova 2 valores extremos da coluna tempo_embarque.

* O IC muda muito? Como interpretar?

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
dados = pd.read_excel("/content/dados_aeroporto (1).xlsx")
dados.head()

passageiro_id     voo companhia  tempo_embarque estado
0              1  REC789       GOL            8.95     PE
1              2  REC123     LATAM            6.08     PE
2              3  REC789     LATAM           17.87     PE
3              4  REC789       GOL           11.31     PE
4              5  REC123       GOL            7.27     PE

In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   passageiro_id   40 non-null     int64 
 1   voo             40 non-null     object
 2   companhia       40 non-null     object
 3   tempo_embarque  40 non-null     object
 4   estado          40 non-null     object
dtypes: int64(1), object(4)
memory usage: 1.7+ KB


In [ ]:
dados['tempo_embarque'] = pd.to_numeric(dados['tempo_embarque'])

In [ ]:
media_amostra = dados['tempo_embarque'].mean()   #média
s_amostra = dados['tempo_embarque'].std()        #desvio
n = len(dados)                                   #tamanho da amostra
gl = n - 1

print(f'Média: {media_amostra:.2f} Desvio padrão: {s_amostra:.2f} Tamanho da amostra: {n}')

Média: 14.01 Desvio padrão: 5.24 Tamanho da amostra: 40


In [ ]:
nc1 = 0.90  #nivel de confiança de 90%

intervalo1 = stats.t.interval(confidence = nc1,                                     #nível de confiança
                              df = gl,                                              #grau de liberdade
                              loc = media_amostra,                                  #media da amostra
                              scale = stats.sem(dados['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra:.2f}")
print(f'Intervalo de confiança de 90%: [{intervalo1[0]:.2f}, {intervalo1[1]:.2f}]')

Média: 14.01
Intervalo de confiança de 90%: [12.61, 15.41]


In [ ]:
nc1 = 0.95  #nivel de confiança de 95%

intervalo2 = stats.t.interval(confidence = nc1,                                     #nível de confiança
                              df = gl,                                              #grau de liberdade
                              loc = media_amostra,                                  #media da amostra
                              scale = stats.sem(dados['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra:.2f}")
print(f'Intervalo de confiança de 95%: [{intervalo2[0]:.2f}, {intervalo2[1]:.2f}]')

Média: 14.01
Intervalo de confiança de 95%: [12.33, 15.69]


In [ ]:
nc1 = 0.99  #nivel de confiança de 99%

intervalo3 = stats.t.interval(confidence = nc1,                                     #nível de confiança
                              df = gl,                                              #grau de liberdade
                              loc = media_amostra,                                  #media da amostra
                              scale = stats.sem(dados['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra:.2f}")
print(f'Intervalo de confiança de 99%: [{intervalo3[0]:.2f}, {intervalo3[1]:.2f}]')

Média: 14.01
Intervalo de confiança de 99%: [11.76, 16.25]


É possível observar que o valor 15 está dentro de todos intervalos de confiança testados (90%, 95%, 99%) o que sugere que o valor 15 é um valor plausível para uma verdadeira média da população.

Também é possível notar que existe uma tendência de aumento do intervalo dos dados a medida que aumentamos o nível de confiança do teste. Isso revela que se eu reduzo a precisão do meu teste eu aumento o intervalo para garantir que o valor procurado está ali dentro, no caso contrário ao aumentar o nível de confiança, eu encontro mais segurança, aceitando um intervalo mais largo e uma estimativa menos precisa.

In [ ]:
dados_ordenados = dados.sort_values(by = "tempo_embarque")

dados_sem_extremos = dados_ordenados.iloc[1:-1]

dados_sem_extremos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 38 entries, 14 to 26
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   passageiro_id   38 non-null     int64  
 1   voo             38 non-null     object 
 2   companhia       38 non-null     object 
 3   tempo_embarque  38 non-null     float64
 4   estado          38 non-null     object 
dtypes: float64(1), int64(1), object(3)
memory usage: 1.8+ KB


In [ ]:
media_amostra2 = dados_sem_extremos['tempo_embarque'].mean()   #média
s_amostra2 = dados_sem_extremos['tempo_embarque'].std()        #desvio
n2 = len(dados_sem_extremos)                                   #tamanho da amostra
g2 = n2 - 1

print(f'Média: {media_amostra2:.2f} Desvio padrão: {s_amostra2:.2f} Tamanho da amostra: {n2}')

Média: 13.77 Desvio padrão: 4.42 Tamanho da amostra: 38


In [ ]:
nc1 = 0.90  #nivel de confiança de 90%

intervalo4 = stats.t.interval(confidence = nc1,                                     #nível de confiança
                              df = g2,                                              #grau de liberdade
                              loc = media_amostra2,                                  #media da amostra
                              scale = stats.sem(dados_sem_extremos['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra2:.2f}")
print(f'Intervalo de confiança de 90%: [{intervalo4[0]:.2f}, {intervalo4[1]:.2f}]')

Média: 13.77
Intervalo de confiança de 90%: [12.56, 14.99]


In [ ]:
nc2 = 0.95  #nivel de confiança de 95%

intervalo5 = stats.t.interval(confidence = nc2,                                                  #nível de confiança
                              df = g2,                                                           #grau de liberdade
                              loc = media_amostra2,                                              #media da amostra
                              scale = stats.sem(dados_sem_extremos['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra2:.2f}")
print(f'Intervalo de confiança de 95%: [{intervalo5[0]:.2f}, {intervalo5[1]:.2f}]')

Média: 13.77
Intervalo de confiança de 95%: [12.32, 15.23]


In [ ]:
nc3 = 0.99  #nivel de confiança de 99%

intervalo6 = stats.t.interval(confidence = nc3,                                                  #nível de confiança
                              df = g2,                                                           #grau de liberdade
                              loc = media_amostra2,                                              #media da amostra
                              scale = stats.sem(dados_sem_extremos['tempo_embarque']))           #erro padrão da média (s/sqtr(n))

print(f"Média: {media_amostra2:.2f}")
print(f'Intervalo de confiança de 99%: [{intervalo6[0]:.2f}, {intervalo6[1]:.2f}]')

Média: 13.77
Intervalo de confiança de 99%: [11.83, 15.72]


**Amostra completa | n=40**

Média: 14.01 | Desvio padrão: 5.24

*   Intervalo de confiança de 90%: [12.61, 15.41]
*   Intervalo de confiança de 95%: [12.33, 15.69]
*   Intervalo de confiança de 99%: [11.76, 16.25]


**Amostra sem os extremos | n=38**

Média: 13.77 | Desvio padrão: 4.42

*   Intervalo de confiança de 90%: [12.56, 14.99]
*   Intervalo de confiança de 95%: [12.32, 15.23]
*   Intervalo de confiança de 99%: [11.83, 15.72]



Retirando os extremos foi possível observar que a média e o desvio da amostra diminuiram, e que os intervalos de confiança também ficaram mais estreitos. Isso reflete o efeito da retirada dos valores extremos, trazendo uma nova estimativa da média de tempo de espera, provavelmente mais precisa, já que a presença desses valores extremos provocavam uma maior variabilidade na amostra.





---

